crates/block-producer/src/server.rs

```diff
@@ -66,8 +66,9 @@ impl BlockProducer {
 
         let rpc_listener = config
             .endpoint
-            .to_socket_addrs()
+            .socket_addrs(|| None)
```

Comment:
Same comment as the faucet main.rs one - this should work but not certain at this stage